client/src/templates/Challenges/utils/worker-executor.js

```diff
@@ -129,16 +129,23 @@ const eventify = self => {
 
   self.emit = (event, ...args) => {
     if (typeof self._events[event] !== 'undefined') {
-      self._events[event].forEach(listener => {
+      let length = self._events[event].length;
+      for (let i = 0; i < length; i++) {
+        const listener = self._events[event][i];
```

Comment:
```suggestion
      const listeners = self._events[event].slice();
      for (let listener of listeners) {
```
The real issue is the splice, but we can work around it by copying the listener array.